### DISCUSSION TITLE
Meaning of "priority" in attribute aware simplification

### DISCUSSION THREAD

**Discussion by JesseRMeyer** [zeux/meshoptimizer#572] [General]
Hi,

Wrt the new attribute aware simplifier: "the weights determine relative priority of attributes between each other and wrt position"

Does priority refer to the relative probability of edge collapse or edge _preservation_?

Also, the documentation give advice regarding [0, 1] inputs, which is of course easy for UV data, but what about [-1, 1] (say, normals), or even [0, N] (say, RGB vertex colors)?

Excellent work, btw.  Thanks for sharing!

Jesse

**Comment by zeux** [comment]
A higher weight assigned to an attribute means that its relative importance grows and as such the edges are less likely to be collapsed if it would lead to a higher attribute delta. I'll look at clarifying the comments.

The precise range is a little less important compared to the general scale of values; eg UV and normal data should not require dramatically different weights (as a side note, UV data is not very useful to feed into the attribute aware simplifier as it's usually just a planar map on the geometry; the normal/color data on the other hand is likely to be significantly distorted for some meshes without this). For RGB vertex colors that are unnormalized (0-255), you'd want to use weights that include a normalization factor, eg `1.0/255.0 * 1e-2` (see https://github.com/zeux/meshoptimizer/blob/master/demo/simplify.html#L127).

It should be noted that the attribute weights are not fully settled in that I'd like to refine the metric that the code uses currently, just lacking solid ideas atm, but this means that it's possible that the behavior of different weights in the future changes as the algorithm gets developed further. Unlike distance errors, where the result is easily interpretable, the attribute weights/errors are currently not, sadly.

**Comment by JesseRMeyer** [comment]
Thanks.  

Side note: My use for supplying UVs was to see the effect of simplification between UV islands where those boundary edges do not have split normals.
